### ISSUE TITLE
Javadoc is broken and Travis did not detect it

### ISSUE THREAD

**Issue by magicDGS** [magicDGS/ReadTools#280]
Error in javadoc generation was introduced in cd0476e6658e431afe2fbd85158fad69f0beb29c. This error was undetected by the Travis build, even if we are running the javadoc task. We should:

* Change the way of testing in Travis if javadoc is working
* Fix the errors in the javadoc generation

The output from `./gradlew clean javadoc` is the following:

```
:clean
:compileJavaNote: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

:processResources
:classes
:javadoc/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/ReadToolsProgram.java:100: warning: no @return
    public final String getToolName() {
                        ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/ReadToolsWalker.java:219: warning: no @return
    public final File getReferenceFile() {
                      ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/ReadToolsWalker.java:224: warning: no @return
    public final boolean isPaired() {
                         ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/RTDataSource.java:167: warning: no @param for factory
    public static void setReadReaderFactory(final ReadReaderFactory factory) {
                       ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/RTDataSource.java:172: warning: no @return
    public boolean isPaired() {
                   ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/RTDataSource.java:177: warning: no @return
    public FastqQualityFormat getOriginalQualityEncoding() {
                              ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/RTDataSource.java:198: warning: no @return
    public SAMFileHeader getHeader() {
                         ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/Main.java:78: warning: no @param for args
    public static void main(final String[] args) {
                       ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/Main.java:92: warning: no @param for args
    protected boolean printOnlyVersion(final String[] args) {
                      ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/Main.java:92: warning: no @return
    protected boolean printOnlyVersion(final String[] args) {
                      ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/RTHelpConstants.java:119: warning: no @param for groupName
    public static String getSuperCategoryProperty(final String groupName) {
                         ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/RTHelpConstants.java:119: warning: no @return
    public static String getSuperCategoryProperty(final String groupName) {
                         ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/read/transformer/barcodes/FixRawBarcodeTagsReadTransformer.java:67: warning: no @param for tagsToUse
    public FixRawBarcodeTagsReadTransformer(final List<String> tagsToUse) {
           ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/read/transformer/barcodes/FixRawBarcodeTagsReadTransformer.java:75: warning: no @param for tagsToUse
    public FixRawBarcodeTagsReadTransformer(final List<String> tagsToUse,
           ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/read/transformer/barcodes/FixRawBarcodeTagsReadTransformer.java:75: warning: no @param for qualityTags
    public FixRawBarcodeTagsReadTransformer(final List<String> tagsToUse,
           ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/tools/trimming/TrimAndFilterPipeline.java:160: warning: no @return
    public List<TrimmerMetric> getTrimmingStats() {
                               ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/tools/trimming/TrimAndFilterPipeline.java:165: warning: no @return
    public List<FilterMetric> getFilterStats() {
                              ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/distmap/DistmapEncoder.java:81: warning: no @return
    public static final String encode(final GATKRead read) {
                               ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/distmap/DistmapEncoder.java:98: warning: no @return
    public static final String encode(final Tuple2<GATKRead, GATKRead> pair) {
                               ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/distmap/DistmapException.java:44: warning: no @param for condition
    public static void distmapValidation(final boolean condition, final Supplier<String> msg) {
                       ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/distmap/DistmapException.java:44: warning: no @param for msg
    public static void distmapValidation(final boolean condition, final Supplier<String> msg) {
                       ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/fastq/FastqGATKWriter.java:49: warning: no @param for writer
    public FastqGATKWriter(final FastqWriter writer) {
           ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/metrics/FilterMetric.java:49: warning: no @param for filter
    public FilterMetric(final String filter) {
           ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/metrics/TrimmerMetric.java:55: warning: no @param for trimmer
    public TrimmerMetric(final String trimmer) {
           ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/tools/barcodes/dictionary/decoder/BarcodeDecoder.java:165: warning: no @return
    public BarcodeDictionary getDictionary() {
                             ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/tools/barcodes/dictionary/decoder/BarcodeDecoder.java:318: warning: no @return
    public MetricsFile<MatcherStat, Integer> getMatcherStatMetrics() {
                                             ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/tools/barcodes/dictionary/decoder/BarcodeDecoder.java:339: warning: no @return
    public MetricsFile<BarcodeStat, Integer> getBarcodeStatMetrics() {
                                             ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/tools/barcodes/dictionary/decoder/BarcodeDecoder.java:102: warning: no @param for maxN
    public BarcodeDecoder(final BarcodeDictionary dictionary, final int maxN,
           ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/tools/barcodes/dictionary/decoder/BarcodeMatch.java:83: warning: no @return
    public String getBarcode() {
                  ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/tools/barcodes/dictionary/decoder/BarcodeMatch.java:98: warning: no @return
    public int getNumberOfNs() {
               ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/tools/barcodes/dictionary/decoder/BarcodeMatch.java:103: warning: no @return
    public int getMismatches() {
               ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/tools/barcodes/dictionary/decoder/BarcodeMatch.java:131: warning: no @return
    public boolean isAmbiguous() {
                   ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/read/writer/ReadToolsIOFormat.java:48: warning: no @return
    public String getExtension();
                  ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/read/writer/ReadToolsIOFormat.java:56: warning: no @param for sourceName
    public default boolean isAssignable(final String sourceName) {
                           ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/read/writer/ReadToolsIOFormat.java:56: warning: no @return
    public default boolean isAssignable(final String sourceName) {
                           ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/FixBarcodeAbstractArgumentCollection.java:75: warning: no @return
    protected abstract List<String> getRawBarcodeQualityTags();
                                    ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/FixBarcodeAbstractArgumentCollection.java:109: warning: no @param for read1
    protected void fixBarcodeForPair(final GATKRead read1, final GATKRead read2) {
                   ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/FixBarcodeAbstractArgumentCollection.java:109: warning: no @param for read2
    protected void fixBarcodeForPair(final GATKRead read1, final GATKRead read2) {
                   ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/FixBarcodeAbstractArgumentCollection.java:173: warning: no @return
    public static final FixBarcodeAbstractArgumentCollection getArgumentCollection(
                                                             ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/ReadGroupArgumentCollection.java:73: warning: no @param for id
    public SAMReadGroupRecord getReadGroupFromArguments(final String id, final String sampleName) {
                              ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/ReadGroupArgumentCollection.java:73: warning: no @param for sampleName
    public SAMReadGroupRecord getReadGroupFromArguments(final String id, final String sampleName) {
                              ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/ReadGroupArgumentCollection.java:73: warning: no @return
    public SAMReadGroupRecord getReadGroupFromArguments(final String id, final String sampleName) {
                              ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTInputArgumentCollection.java:89: warning: no @return
    public RTDataSource getDataSource(final File referenceFileName) {
                        ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTOutputArgumentCollection.java:65: warning: no @return
    public ReadWriterFactory getWriterFactory() {
                             ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTOutputArgumentCollection.java:85: warning: no @return
    public final GATKReadWriter outputWriter(final SAMFileHeader header,
                                ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTOutputArgumentCollection.java:96: warning: no @param for suffix
    public abstract String getOutputNameWithSuffix(final String suffix);
                           ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTOutputArgumentCollection.java:96: warning: no @return
    public abstract String getOutputNameWithSuffix(final String suffix);
                           ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTOutputArgumentCollection.java:102: warning: no @param for suffix
    public abstract Path makeMetricsFile(final String suffix);
                         ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTOutputArgumentCollection.java:102: warning: no @return
    public abstract Path makeMetricsFile(final String suffix);
                         ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTOutputArgumentCollection.java:119: warning: no @return
    protected abstract GATKReadWriter createWriter(final ReadWriterFactory factory,
                                      ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTOutputArgumentCollection.java:123: warning: no @param for header
    protected abstract void updateHeader(final SAMFileHeader header,
                            ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTOutputArgumentCollection.java:123: warning: no @param for programRecord
    protected abstract void updateHeader(final SAMFileHeader header,
                            ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTOutputArgumentCollection.java:127: warning: no @return
    public static final RTOutputArgumentCollection defaultOutput() {
                                                   ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTOutputArgumentCollection.java:132: warning: no @return
    public static final RTOutputArgumentCollection fastqOutput() {
                                                   ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTOutputArgumentCollection.java:137: warning: no @return
    public static final RTOutputArgumentCollection splitOutput() {
                                                   ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/cmd/argumentcollections/RTAbstractOutputBamArgumentCollection.java:67: warning: no @param for outputName
    protected final void validateUserOutput(final String outputName) {
                         ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/ReadsSourceHandler.java:67: warning: no @return
    public String getHandledSource() {
                  ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/ReadsSourceHandler.java:72: warning: no @param for maxNumberOfReads
    public abstract FastqQualityFormat getQualityEncoding(final long maxNumberOfReads);
                                       ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/ReadsSourceHandler.java:72: warning: no @return
    public abstract FastqQualityFormat getQualityEncoding(final long maxNumberOfReads);
                                       ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/ReadsSourceHandler.java:75: warning: no @return
    public abstract SAMFileHeader getHeader();
                                  ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/ReadsSourceHandler.java:81: warning: no @return
    public abstract Iterator<GATKRead> toIterator();
                                       ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/ReadsSourceHandler.java:90: warning: no @param for locs
    public abstract Iterator<GATKRead> toIntervalIterator(final List<SimpleInterval> locs);
                                       ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/ReadsSourceHandler.java:90: warning: no @return
    public abstract Iterator<GATKRead> toIntervalIterator(final List<SimpleInterval> locs);
                                       ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/ReadsSourceHandler.java:93: warning: no @return
    public Stream<GATKRead> toStream() {
                            ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/ReadsSourceHandler.java:108: warning: no @param for factory
    public static ReadsSourceHandler getHandler(final String source,
                                     ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/FastqSourceHandler.java:53: warning: no @param for factory
    public FastqSourceHandler(final String source, final ReadReaderFactory factory) {
           ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/FileSourceHandler.java:79: warning: no @param for reader
    protected abstract SAMFileHeader getReaderHeader(final T reader);
                                     ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/FileSourceHandler.java:79: warning: no @return
    protected abstract SAMFileHeader getReaderHeader(final T reader);
                                     ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/FileSourceHandler.java:89: warning: no @param for reader
    protected abstract Iterator<GATKRead> getReaderIntervalIterator(final T reader,
                                          ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/FileSourceHandler.java:89: warning: no @param for locs
    protected abstract Iterator<GATKRead> getReaderIntervalIterator(final T reader,
                                          ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/FileSourceHandler.java:89: warning: no @return
    protected abstract Iterator<GATKRead> getReaderIntervalIterator(final T reader,
                                          ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/FileSourceHandler.java:86: warning: no @param for reader
    protected abstract Iterator<GATKRead> getReaderIterator(final T reader);
                                          ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/FileSourceHandler.java:86: warning: no @return
    protected abstract Iterator<GATKRead> getReaderIterator(final T reader);
                                          ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/FileSourceHandler.java:82: warning: no @param for reader
    protected abstract FastqQualityFormat getReaderQualityEncoding(final T reader,
                                          ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/FileSourceHandler.java:82: warning: no @param for maxNumberOfReads
    protected abstract FastqQualityFormat getReaderQualityEncoding(final T reader,
                                          ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/FileSourceHandler.java:82: warning: no @return
    protected abstract FastqQualityFormat getReaderQualityEncoding(final T reader,
                                          ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/engine/sourcehandler/ReadsSourceHandler.java:62: warning: no @param for source
    protected ReadsSourceHandler(final String source) {
              ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/documentation/RTGSONWorkUnit.java:42: warning: no @param for note
    public void setNote(final String note) {
                ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/documentation/RTGSONWorkUnit.java:47: warning: no @param for warning
    public void setWarning(final String warning) {
                ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/documentation/RTHelpDoclet.java:67: warning: no @param for rootDoc
    public static boolean start(final RootDoc rootDoc) throws IOException {
                          ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/documentation/RTHelpDoclet.java:67: warning: no @return
    public static boolean start(final RootDoc rootDoc) throws IOException {
                          ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/documentation/RTHelpDoclet.java:67: warning: no @throws for java.io.IOException
    public static boolean start(final RootDoc rootDoc) throws IOException {
                          ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/read/transformer/trimming/TrimmingFunction.java:106: warning: no @param for disable5prime
    public final void setDisableEnds(final boolean disable5prime, final boolean disable3prime) {
                      ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/read/transformer/trimming/TrimmingFunction.java:106: warning: no @param for disable3prime
    public final void setDisableEnds(final boolean disable5prime, final boolean disable3prime) {
                      ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/read/transformer/trimming/TrimmingFunction.java:118: warning: no @return
    public final boolean isDisable5prime() {
                         ^
/Users/daniel/workspaces/ReadTools/src/main/java/org/magicdgs/readtools/utils/read/transformer/trimming/TrimmingFunction.java:123: warning: no @return
    public final boolean isDisable3prime() {
                         ^
javadoc: error - com.sun.tools.doclets.internal.toolkit.util.DocletAbortException: java.lang.ClassCastException: com.sun.tools.javadoc.MethodDocImpl cannot be cast to com.sun.tools.javadoc.AnnotationTypeElementDocImpl

1 error
86 warnings
:javadoc FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javadoc'.
> Javadoc generation failed. Generated Javadoc options file (useful for troubleshooting): '/Users/daniel/workspaces/ReadTools/build/tmp/javadoc/javadoc.options'

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 5.947 secs

```

**Comment by magicDGS** [comment]
This is an error in Java's javadoc code, but it could be solved by not referring to a default value of an annotation.

**Comment by magicDGS** [comment]
Closed by #281
